File: api_test_platform/frontend/src/types/environment.ts
/**
 * 环境管理模块类型定义
 */

// 环境接口
export interface Environment {
  id: number;
  team_id: number;
  name: string;
  description?: string;
  is_default: boolean;
  created_by: number;
  created_at: string;
  updated_at?: string;
  service_count?: number;
  services?: Service[];
  variables?: EnvVariable[];
}

// 环境创建请求
export interface EnvironmentCreate {
  name: string;
  description?: string;
}

// 环境更新请求
export interface EnvironmentUpdate {
  name?: string;
  description?: string;
}

// 服务接口
export interface Service {
  id: number;
  environment_id: number;
  name: string;
  service_type?: 'microservice' | 'module' | 'third_party';
  description?: string;
  sort_order: number;
  created_at: string;
  updated_at?: string;
  servers?: Server[];
  databases?: Database[];
}

// 服务创建请求
export interface ServiceCreate {
  name: string;
  service_type?: 'microservice' | 'module' | 'third_party';
  description: string;
}

// 服务更新请求
export interface ServiceUpdate {
  name?: string;
  service_type?: 'microservice' | 'module' | 'third_party';
  description?: string;
}

// 服务器配置接口
export interface Server {
  id: number;
  service_id: number;
  name: string;
  host: string;
  port?: number;
  protocol: 'http' | 'https';
  base_path?: string;
  ssl_config?: Record<string, any>;
  headers?: Record<string, string>;
  timeout: number;
  base_url: string;
  created_at: string;
  updated_at?: string;
}

// 服务器创建请求
export interface ServerCreate {
  name?: string;
  host: string;
  port?: number;
  protocol?: 'http' | 'https';
  base_path?: string;
  ssl_config?: Record<string, any>;
  headers?: Record<string, string>;
  timeout?: number;
}

// 服务器更新请求
export interface ServerUpdate {
  name?: string;
  host?: string;
  port?: number;
  protocol?: 'http' | 'https';
  base_path?: string;
  ssl_config?: Record<string, any>;
  headers?: Record<string, string>;
  timeout?: number;
}

// 数据库配置接口
export interface Database {
  id: number;
  service_id: number;
  name: string;
  db_type: 'mysql' | 'postgresql' | 'mongodb' | 'redis' | 'sqlserver' | 'oracle';
  host: string;
  port?: number;
  database?: string;
  username?: string;
  password?: string;
  charset?: string;
  extra_config?: Record<string, any>;
  created_at: string;
  updated_at?: string;
}

// 数据库创建请求
export interface DatabaseCreate {
  name: string;
  db_type: 'mysql' | 'postgresql' | 'mongodb' | 'redis' | 'sqlserver' | 'oracle';
  host: string;
  port?: number;
  database?: string;
  username?: string;
  password?: string;
  charset?: string;
  extra_config?: Record<string, any>;
  is_encrypted?: boolean;
}

// 数据库更新请求
export interface DatabaseUpdate {
  name?: string;
  db_type?: 'mysql' | 'postgresql' | 'mongodb' | 'redis' | 'sqlserver' | 'oracle';
  host?: string;
  port?: number;
  database?: string;
  username?: string;
  password?: string;
  charset?: string;
  extra_config?: Record<string, any>;
}

// 环境变量接口
export interface EnvVariable {
  id: number;
  environment_id: number;
  key: string;
  value?: string;
  value_type: 'string' | 'number' | 'boolean';
  is_encrypted: boolean;
  description?: string;
  created_at: string;
  updated_at?: string;
}

// 环境变量创建请求
export interface EnvVariableCreate {
  key: string;
  value?: string;
  value_type?: 'string' | 'number' | 'boolean';
  is_encrypted?: boolean;
  description?: string;
}

// 环境变量更新请求
export interface EnvVariableUpdate {
  key?: string;
  value?: string;
  value_type?: 'string' | 'number' | 'boolean';
  is_encrypted?: boolean;
  description?: string;
}

// 分页响应
export interface PaginatedResponse<T> {
  items: T[];
  total: number;
  page: number;
  page_size: number;
}

// 数据库类型选项
export const DATABASE_TYPES = [
  { value: 'mysql', label: 'MySQL', defaultPort: 3306 },
  { value: 'postgresql', label: 'PostgreSQL', defaultPort: 5432 },
  { value: 'mongodb', label: 'MongoDB', defaultPort: 27017 },
  { value: 'redis', label: 'Redis', defaultPort: 6379 },
  { value: 'sqlserver', label: 'SQL Server', defaultPort: 1433 },
  { value: 'oracle', label: 'Oracle', defaultPort: 1521 }
] as const;

// 服务类型选项
export const SERVICE_TYPES = [
  { value: 'microservice', label: '微服务' },
  { value: 'module', label: '应用模块' },
  { value: 'third_party', label: '第三方服务' }
] as const;

// 变量类型选项
export const VARIABLE_TYPES = [
  { value: 'string', label: 'String' },
  { value: 'number', label: 'Number' },
  { value: 'boolean', label: 'Boolean' }
] as const;

File: api_test_platform/backend/app/schemas/environment.py
"""
Environment Management Schemas
"""
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from enum import Enum


class ServiceType(str, Enum):
    MICROSERVICE = "microservice"
    MODULE = "module"
    THIRD_PARTY = "third_party"


class ValueType(str, Enum):
    STRING = "string"
    NUMBER = "number"
    BOOLEAN = "boolean"


class ServerBase(BaseModel):
    name: str = Field(..., max_length=100)
    host: str = Field(..., max_length=255)
    port: Optional[int] = None
    protocol: str = Field(default="https", max_length=20)
    base_path: str = Field(default="", max_length=255)
    ssl_config: Optional[str] = None
    headers: Optional[str] = None
    timeout: int = Field(default=30000)
    is_default: bool = False


class ServerCreate(ServerBase):
    pass


class ServerUpdate(BaseModel):
    name: Optional[str] = Field(None, max_length=100)
    host: Optional[str] = Field(None, max_length=255)
    port: Optional[int] = None
    protocol: Optional[str] = Field(None, max_length=20)
    base_path: Optional[str] = Field(None, max_length=255)
    ssl_config: Optional[str] = None
    headers: Optional[str] = None
    timeout: Optional[int] = None
    is_default: Optional[bool] = None


class ServerResponse(ServerBase):
    id: int
    service_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class DatabaseBase(BaseModel):
    name: str = Field(..., max_length=100)
    db_type: str = Field(..., max_length=50)
    host: str = Field(..., max_length=255)
    port: Optional[int] = None
    database: str = Field(..., max_length=100)
    username: str = Field(..., max_length=100)
    password: str = Field(..., max_length=255)
    charset: str = Field(default="utf8mb4", max_length=20)
    extra_config: Optional[str] = None
    is_default: bool = False


class DatabaseCreate(DatabaseBase):
    pass


class DatabaseUpdate(BaseModel):
    name: Optional[str] = Field(None, max_length=100)
    db_type: Optional[str] = Field(None, max_length=50)
    host: Optional[str] = Field(None, max_length=255)
    port: Optional[int] = None
    database: Optional[str] = Field(None, max_length=100)
    username: Optional[str] = Field(None, max_length=100)
    password: Optional[str] = Field(None, max_length=255)
    charset: Optional[str] = Field(None, max_length=20)
    extra_config: Optional[str] = None
    is_default: Optional[bool] = None


class DatabaseResponse(BaseModel):
    id: int
    service_id: int
    name: str
    db_type: str
    host: str
    port: Optional[int]
    database: str
    username: str
    password: str
    charset: str
    extra_config: Optional[str]
    is_default: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ServiceBase(BaseModel):
    name: str = Field(..., max_length=100)
    service_type: ServiceType = ServiceType.MICROSERVICE
    description: Optional[str] = None
    sort_order: int = 0


class ServiceCreate(ServiceBase):
    pass


class ServiceUpdate(BaseModel):
    name: Optional[str] = Field(None, max_length=100)
    service_type: Optional[ServiceType] = None
    description: Optional[str] = None
    sort_order: Optional[int] = None


class ServiceResponse(ServiceBase):
    id: int
    environment_id: int
    created_at: datetime
    updated_at: datetime
    servers: List[ServerResponse] = []
    databases: List[DatabaseResponse] = []

    class Config:
        from_attributes = True


class ServiceListResponse(BaseModel):
    items: List[ServiceResponse]
    total: int
    page: int
    page_size: int


class VariableBase(BaseModel):
    key: str = Field(..., max_length=100)
    value: Optional[str] = None
    value_type: ValueType = ValueType.STRING
    is_encrypted: bool = False
    description: Optional[str] = None


class VariableCreate(VariableBase):
    pass


class VariableUpdate(BaseModel):
    key: Optional[str] = Field(None, max_length=100)
    value: Optional[str] = None
    value_type: Optional[ValueType] = None
    is_encrypted: Optional[bool] = None
    description: Optional[str] = None


class VariableResponse(VariableBase):
    id: int
    environment_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class EnvironmentBase(BaseModel):
    name: str = Field(..., max_length=100)
    description: Optional[str] = None
    is_default: bool = False


class EnvironmentCreate(EnvironmentBase):
    team_id: Optional[int] = None
    services: Optional[List[ServiceCreate]] = None
    variables: Optional[List[VariableCreate]] = None


class EnvironmentUpdate(BaseModel):
    name: Optional[str] = Field(None, max_length=100)
    description: Optional[str] = None
    is_default: Optional[bool] = None
    is_active: Optional[bool] = None


class EnvironmentResponse(EnvironmentBase):
    id: int
    team_id: Optional[int]
    is_active: bool
    created_by: Optional[int]
    created_at: datetime
    updated_at: datetime
    services: List[ServiceResponse] = []
    variables: List[VariableResponse] = []

    class Config:
        from_attributes = True


class EnvironmentListResponse(BaseModel):
    items: List[EnvironmentResponse]
    total: int
    page: int
    page_size: int


class EnvironmentSummary(BaseModel):
    id: int
    name: str
    description: Optional[str]
    is_default: bool
    is_active: bool
    service_count: int
    variable_count: int

    class Config:
        from_attributes = True


class VariableResolveRequest(BaseModel):
    environment_id: int
    text: str
    case_variables: Optional[dict] = None
    scenario_variables: Optional[dict] = None


class VariableResolveResponse(BaseModel):
    original_text: str
    resolved_text: str
    resolved_variables: dict
    unresolved_variables: List[str]
